### PR TITLE
vim-patch:8.2.1078: highlight and match functionality together in one file

### PR DESCRIPTION
**Pull request by lewis6991**
Problem:    Highlight and match functionality together in one file.
Solution:   Move match functionality to a separate file. (Yegappan Lakshmanan,
            closes vim/vim#6352)

https://github.com/vim/vim/commit/06cf97e714fd8bf9b35ff5f8a6f2302c79acdd03

### CODE REVIEW

**Comment by zeertzjq** [comment]
Commit message is truncated.

File: src/nvim/match.c [hunk 1]
```diff
@@ -0,0 +1,1179 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
```
Comment (zeertzjq) [inline src/nvim/match.c:3]: Missing `match.c: functions for highlighting matches` comment.

**Comment by lewis6991** [comment]
> Commit message is truncated.

What do you mean, and what is the fix?

**Comment by zeertzjq** [comment]
I mean the commit body: `Lakshm>`

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**